.github/workflows/publish-release.yml

```diff
@@ -15,33 +22,101 @@ jobs:
+      - uses: MetaMask/action-publish-release@v3
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - run: yarn --frozen-lockfile
-      - name: Get release version
-        id: version
+      - name: Install
         run: |
-          PREFIX='release/'
-          echo ::set-output name=NEW_VERSION::${GITHUB_HEAD_REF#"$PREFIX"}
-      - run: yarn build
-      - name: Deploy Version
-        uses: peaceiris/actions-gh-pages@068dc23d9710f1ba62e86896f84735d869951305
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir:  dist/build
-          destination_dir: ${{ steps.version.outputs.NEW_VERSION }}
-      - name: Deploy Latest
-        uses: peaceiris/actions-gh-pages@068dc23d9710f1ba62e86896f84735d869951305
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir:  dist/build
-          destination_dir: latest
-
+          yarn install
+          yarn build
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: |
+            ./dist
+            ./node_modules/.yarn-state.yml
+          key: ${{ github.sha }}
+  publish-npm-dry-run:
+    runs-on: ubuntu-latest
+    needs: publish-release
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.sha }}
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: |
+            ./dist
+            ./node_modules/.yarn-state.yml
+          key: ${{ github.sha }}
+      - name: Dry Run Publish
+        # omit npm-token token to perform dry run publish
+        uses: MetaMask/action-npm-publish@v4
+        with:
+          slack-webhook-url: ${{ secrets.SLACK_WEBHOOK_URL }}
+          subteam: S042S7RE4AE # @metamask-npm-publishers
+        env:
+          SKIP_PREPACK: true
+  publish-npm:
+    environment: npm-publish
+    runs-on: ubuntu-latest
+    needs: publish-npm-dry-run
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.sha }}
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: |
+            ./dist
+            ./node_modules/.yarn-state.yml
+          key: ${{ github.sha }}
+      - name: Publish
+        uses: MetaMask/action-npm-publish@v2
+        with:
+          # This `NPM_TOKEN` needs to be manually set per-repository.
+          # Look in the repository settings under "Environments", and set this token in the `npm-publish` environment.
+          npm-token: ${{ secrets.NPM_TOKEN }}
+        env:
+          SKIP_PREPACK: true
+  get-release-version:
+    runs-on: ubuntu-latest
+    needs: publish-npm
+    outputs:
+      RELEASE_VERSION: ${{ steps.get-release-version.outputs.RELEASE_VERSION }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.sha }}
+      - id: get-release-version
+        shell: bash
+        run: ./scripts/get.sh ".version" "RELEASE_VERSION"
```

Comment:
Looks like this script is missing

Reply:
The `prepack.sh` script is missing as well. That script is used to conditionally skip the pre-pubish build step (in this workflow, when we're publishing we've already built it).

It's called `prepack` because that's the name of the pre-publish lifecycle script in Yarn v2+, but on Yarn v1 we'd need to call that in `prepublishOnly`

Reply:
Added in https://github.com/MetaMask/api-specs/pull/224/commits/ff33bf5ee5745d8d4944d17a2a68a1fd881333b7